static-analysis-plugin/plugin/src/main/groovy/com/novoda/staticanalysis/StaticAnalysisPlugin.groovy

```diff
@@ -1,15 +1,23 @@
 package com.novoda.staticanalysis
 
-import org.gradle.api.Plugin
-import org.gradle.api.Project
+import org.gradle.api.*
 
 class StaticAnalysisPlugin implements Plugin<Project> {
     private final CheckstyleConfigurator checkstyleConfigurator = new CheckstyleConfigurator()
 
     @Override
     void apply(Project project) {
         StaticAnalysisExtension extension = project.extensions.create('staticAnalysis', StaticAnalysisExtension)
-        checkstyleConfigurator.configure(project, extension)
+
+        NamedDomainObjectContainer<Violations> allViolations = project.container(Violations)
+        Task evaluateViolations = project.tasks.create('evaluateViolations', EvaluateViolationsTask) { task ->
+            task.penalty = extension.penalty
```

Comment:
I wonder if this isn't evaluated too early here. Usually, you'd use conventionMapping in cases like this.